### PR TITLE
Performance Profile Creator Documentation

### PR DESCRIPTION
**Pull request by swatisehgal**
Documentation for the PRs:
1. https://github.com/openshift-kni/performance-addon-operators/pull/511
2. https://github.com/openshift-kni/performance-addon-operators/pull/513
3. https://github.com/openshift-kni/performance-addon-operators/pull/515

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

### CODE REVIEW

File: cmd/performance-profile-creator/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+A tool to automate the process of creating Performance Profile using the user supplied profile parameters.
+
+## Software Components
+
+1. A CLI tool part of the PAO image
+1. A wrapper script <!--(TODO: Add link to where the wrapper script is placed) --> automates the access to PPC tool. It pulls the PAO image, supplies arguments and runs the tool inside Performance Addon Operator container with performance-profile-creator entrypoint.
+
+## Flow
+1. The user supplies the command line args to the tool by using a wrapper script.
+1. The wrapper triggers the must-gather tool.
```
Comment (fromanirh) [inline cmd/performance-profile-creator/README.md:12]: it is correct that all the bullet points start with `1.` ?

File: cmd/performance-profile-creator/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+1. A must-gather output file is passed to PPC as an argument.
```
Comment (fromanirh) [inline cmd/performance-profile-creator/README.md:13]: so is not clear if the user can reuse/pass an existing must-gather content to the PPC tool

File: cmd/performance-profile-creator/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+1. The wrapper decompresses the must-gather output file and “mount” the ghw snapshot in order to inspect the hardware, and passes the snapshot to PPC. The ghw snapshots can also be supplied by the user may directly.
```
Comment (fromanirh) [inline cmd/performance-profile-creator/README.md:14]: this is an implementation detail, so depending on the target audience (users vs developers) could be deferred to a later section or to a different document entirely.

From user perspective the most relevant content seems:
1. PPC consumes a must-gather output
2. PPC may run must-gather directly if the must-gather output is not given
3. output is a bunc of YAML data (PAO profile + NTO tuned part)

File: cmd/performance-profile-creator/README.md (hunk 1 above)
Comment (swatisehgal) [reply cmd/performance-profile-creator/README.md:12]: Yep, I think it should be fine as they are rendered as 1, 2, 3 etc., I learnt this recently while writing NFD documentation. Take a look [here](https://guides.github.com/features/mastering-markdown/).

File: cmd/performance-profile-creator/README.md (hunk 2 above)
Comment (swatisehgal) [reply cmd/performance-profile-creator/README.md:13]: Okay, will try to improve this.

File: cmd/performance-profile-creator/README.md (hunk 3 above)
Comment (swatisehgal) [reply cmd/performance-profile-creator/README.md:14]: Perfect, will update as above. Thanks for the suggestion

File: cmd/performance-profile-creator/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+   make create-performance-profile
+   ```
+1. To build the Performance addon Operator image with  from source:
+   ```bash
+   make operator-container
+   ```
+Alternatively, you can pull the latest master upstream image.  In the following examples, TAG has the format major.minor-snapshot. For example, the TAG for OpenShift 4.8 will be 4.8-snapshot:
+
+```bash
+podman pull quay.io/repository/openshift-kni/performance-addon-operator:<TAG>
```
Comment (marcel-apf) [inline cmd/performance-profile-creator/README.md:42]: do we need the "repository" ? also I would use the tag you mentioned above, 4.8-snapshot

File: cmd/performance-profile-creator/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+## Running Performance Profile Creator
+
+To run the Performance profile Creator run:
+
+```bash
+podman run --entrypoint performance-profile-creator -v ./must-gather:/must-gather:z  \
```
Comment (marcel-apf) [inline cmd/performance-profile-creator/README.md:50]: The assumption here is there we have a  must-gather directory available where we run the tool.
It worth mentioning it. Meaning we need to either run must-gather tool and use its output dir or decompress and existing must-gather tarball and use its decompressed dir.
 This is for the first ./must-gather:
For the second one we can leave it /must-gather, but we need to pass -M /must-gather to the tool itself after the image, for example:
podman run --entrypoint performance-profile-creator -v <must-gather-content-dir>:/must-gather:z quay.io/swsehgal/perf-profile-creator:working -M /must-gather

File: cmd/performance-profile-creator/README.md (hunk 4 above)
Comment (swatisehgal) [reply cmd/performance-profile-creator/README.md:42]: Ah! My bad, will update.

File: cmd/performance-profile-creator/README.md (hunk 5 above)
Comment (swatisehgal) [reply cmd/performance-profile-creator/README.md:50]: Will update, thanks!

**Comment by fromanirh** [comment]
/retest
even though not really needed

**Comment by marcel-apf** [comment]
/lgtm
We can continue to add info as we go.

**Pull request merged**